File: Android/app/src/main/java/com/mikefilion/hueapp/Hue/UrlProvider.java
package com.mikefilion.hueapp.Hue;

/**
 * Created by mfilion on 2016-05-02.
 */
public class UrlProvider {

    private String ip;

    public UrlProvider(String ip)
    {
        this.ip = ip;

        if (!this.ip.startsWith("http://"))
            this.ip = "http://" + this.ip;

        if(this.ip.endsWith("/")) {
            this.ip = this.ip.substring(0, this.ip.length() - 1);
        }

        this.ip = this.ip.replace("/description.xml", "");
    }

    public String GetStatusUrl()
    {
        return ip + "/api/" + Settings.GetDefault().GetBridgeApiKey();
    }

    public String GetRegisterUrl()
    {
        return ip + "/api";
    }

    public String GetLampUrl(String lightKey)
    {
        return ip + "/api/" + Settings.GetDefault().GetBridgeApiKey() + "/lights/" + lightKey + "/state";
    }

    public String GetRuleUrl(String ruleKey)
    {
        return ip + "/api/" + Settings.GetDefault().GetBridgeApiKey() + "/rules/" + ruleKey + "/";
    }
}

File: C#/Hue/UrlProvider.cs
using Hue.Properties;

namespace Hue
{
    public class UrlProvider
    {
        private string ip;

        public UrlProvider(string ip)
        {
            this.ip = ip;

            if (!this.ip.StartsWith("http://"))
                this.ip = "http://" + this.ip;

            this.ip = this.ip.TrimEnd('/');
            this.ip = this.ip.Replace("/description.xml","");
        }

        public string GetStatusUrl()
        {
            return ip + "/api/" + Settings.Default.BridgeApiKey;
        }

        public string GetRegisterUrl()
        {
            return ip + "/api";
        }

        public string GetLampUrl(string lightKey)
        {
            return ip + "/api/" + Settings.Default.BridgeApiKey + "/lights/" + lightKey + "/state";
        }

        public string GetRuleUrl(string ruleKey)
        {
            return ip + "/api/" + Settings.Default.BridgeApiKey + "/rules/" + ruleKey + "/";
        }
    }
}
